homeassistant/components/rotel/media_player.py

```diff
@@ -0,0 +1,493 @@
+    """
+    _LOGGER.debug("Setting up hass services")
+
+    async def async_handle_send_command(entity, call):
+        command_name = call.data.get(ATTR_COMMAND_NAME)
+        if isinstance(entity, RotelMediaPlayer):
+            _LOGGER.debug(
+                "%s service sending command %s to entity %s",
+                SERVICE_SEND_COMMAND,
+                command_name,
+                entity.entity_id)
+            await entity.send_command(command_name)
+        else:
+            _LOGGER.debug(
+                "%s service not sending command %s to incompatible entity %s",
+                SERVICE_SEND_COMMAND,
+                command_name,
+                entity.entity_id)
+
+    async def async_handle_reconnect(entity, call):
+        # pylint: disable=unused-argument
+        if isinstance(entity, RotelMediaPlayer):
+            _LOGGER.debug("%s service reconnecting entity %s",
+                          SERVICE_RECONNECT, entity.entity_id)
+            await entity.reconnect()
+        else:
+            _LOGGER.debug("%s service not reconnecting incompatible entity %s",
+                          SERVICE_RECONNECT, entity.entity_id)
+
+    component = hass.data[DOMAIN]
+    component.async_register_entity_service(
+        SERVICE_SEND_COMMAND, SERVICE_SEND_COMMAND_SCHEMA,
+        async_handle_send_command)
+    component.async_register_entity_service(
+        SERVICE_RECONNECT, SERVICE_RECONNECT_SCHEMA,
+        async_handle_reconnect)
+
+
+class RotelMediaPlayer(MediaPlayerDevice):
+    """Representation of a Rotel media player."""
+
+    # pylint: disable=abstract-method
+    # pylint: disable=too-many-public-methods
+    # pylint: disable=too-many-instance-attributes
+
+    def __init__(self, name, device, source_aliases):
+        """Initialize the device."""
+        self._name = name
+        self._device = device
+        self._conn = None  # Make sure that you call open_connection...
+        self._read_messages_task = None  # ... and start_read_messages
+        self._state = STATE_OFF
+        self._source_name = None
+        self._volume = None
+        self.set_source_lists(source_aliases)
+        self._mute_on = None
+        self._party_mode_on = None
+        self._info = None
+        self._icons = None
+        self._speaker_icons = None
+        self._state_icons = None
+        self._input_icons = None
+        self._sound_mode_icons = None
+        self._misc_icons = None
+        self._triggers = None
+
+    @property
+    def should_poll(self) -> bool:
+        """Do not poll because this entity pushes its state to HA."""
+        return False
+
+    async def open_connection(self):
+        """Open a connection to the device."""
+        from rsp1570serial.connection import RotelAmpConn
+        conn = RotelAmpConn(self._device)
+        try:
+            await conn.open()
+        except Exception:
+            _LOGGER.error("Could not open connection", exc_info=True)
```

Comment:
If this is truly unexpected then:

```suggestion
            _LOGGER.exception("Could not open connection")
```